src/components/SpotsMap/component.tsx

```diff
@@ -18,6 +18,7 @@ export const SpotsMap: React.FC<SpotsMapProps> = () => {
   const location = useLocation();
   const spots = useSpots(center);
   const history = useHistory();
+  const listRef = useRef<any>(null);
```

Comment:
We can type the ref here and type guard `listRef` in `useEffect`.

```suggestion
  const listRef = useRef<HTMLUListElement>(null);
```

Reply:
Yep indeed. Its done now. Not really sure its usefull though, as in practice this ref can never be null